src/ert/run_models/everest_run_model.py

```diff
@@ -257,6 +259,54 @@ def _create_optimizer(self) -> BasicOptimizer:
             ),
         )
 
+        def _forward_ropt_event(everest_event: OptimizerEvent):
+            has_results = bool(everest_event.data and everest_event.data.get("results"))
+
+            # The batch these results pertain to
+            # If the event has results, they usually pertain to the
+            # batch before self._batch_id, i.e., self._batch_id - 1
+            batch_id = (
+                everest_event.data["results"][0].batch_id
+                if has_results
+                else self._batch_id
+            )
+
+            data_json = {}
+            if has_results:
+                results = everest_event.data["results"]
+                exit_code = everest_event.data["exit_code"]
+
+                result_types = [
+                    "FunctionResults"
+                    if isinstance(r, FunctionResults)
+                    else "GradientResults"
+                    for r in results
+                ]
+
+                data_json = {
+                    "exit_code": exit_code,
+                    "results": {},
+                }
+
+                if any(isinstance(r, GradientResults) for r in result_types):
+                    data_json["results"]["gradient"] = True
+
+                if any(isinstance(r, FunctionResults) for r in result_types):
+                    data_json["results"]["gradient"] = True
+
+            self.send_event(
```

Comment:
The `send_event` method is expecting a `StatusEvent`, which results in type incompatibility issue that I think can be easily solved by just adding `EverestStatusEvent` to `StatusEvent` in `src/ert/run_models/event.py`

``` python
StatusEvents = (
    AnalysisStatusEvent
    | AnalysisTimeEvent
    | EndEvent
    | EverestStatusEvent <---- new
    | FullSnapshotEvent
    | SnapshotUpdateEvent
    | RunModelErrorEvent
    | RunModelStatusEvent
    | RunModelTimeEvent
    | RunModelUpdateBeginEvent
    | RunModelDataEvent
    | RunModelUpdateEndEvent
)
```